test/sqllogictest/explain/raw_plan_as_text.slt

```diff
@@ -25,51 +25,312 @@ mode cockroach
+#
+# Filter exists(%1)
+# | Get materialize.public.t
+# |
+# | Subquery %1
+# |   Filter (#^0 = #0)
+# |     Get materialize.public.t
+query T multiline
+EXPLAIN RAW PLAN AS TEXT FOR
+SELECT * FROM t WHERE EXISTS(SELECT * FROM t as t1 WHERE t.a = t1.a)
+----
+Filter ???
+  Get materialize.public.t
+
+EOF
+
+query T multiline
+EXPLAIN RAW PLAN AS TEXT FOR
+SELECT t1.a, t2.a FROM t as t1, t as t2
+----
+Project #0, #2
+  InnerJoin true
+    Get materialize.public.t
+    Get materialize.public.t
 
 EOF
+
+query T multiline
+EXPLAIN RAW PLAN AS TEXT FOR
+SELECT t1.a, t2.a FROM t as t1, t as t2 WHERE t1.b = t2.b
+----
+Project #0, #2
+  Filter (#1 = #3)
+    InnerJoin true
+      Get materialize.public.t
+      Get materialize.public.t
+
+EOF
+
+query T multiline
+EXPLAIN RAW PLAN AS TEXT FOR
+SELECT t1.a, t2.a FROM t as t1 JOIN t as t2 ON t1.b = t2.b
+----
+Project #0, #2
+  InnerJoin (#1 = #3)
+    Get materialize.public.t
+    Get materialize.public.t
+
+EOF
+
+query T multiline
+EXPLAIN RAW PLAN AS TEXT FOR
+SELECT t1.a, t2.a FROM t as t1 JOIN t as t2 ON t1.b = t2.b
+----
+Project #0, #2
+  InnerJoin (#1 = #3)
+    Get materialize.public.t
+    Get materialize.public.t
+
+EOF
+
+
+# A case where we cannot pull the let statement up through teh join,
+# because the local l0 is correlated against the lhs of the enclosing join
+query T multiline
+EXPLAIN RAW PLAN AS TEXT FOR
+SELECT
+  *
+FROM
+  (
+    SELECT * FROM t
+  ) as r1
+  CROSS JOIN LATERAL (
+    WITH r2 as (
+      SELECT MAX(r1.a * t.a) AS m FROM t
+    )
+    SELECT * FROM r2 WHERE r2.m != r1.a
+  ) as r3
+  CROSS JOIN LATERAL (
+    WITH r4 as (
+      SELECT MAX(r1.a * t.a) AS m FROM t
+    )
+    SELECT * FROM r4 WHERE r4.m != r1.a
+  ) as r5;
+----
+InnerJoin true
+  InnerJoin true
+    Get materialize.public.t
+    Let
+      Filter (#0 != #^0)
+        Get l0
+      Local l0
+        Reduce aggregates=[max((#^0 * #0))]
+          Get materialize.public.t
+  Let
+    Filter (#0 != #^0)
+      Get l0
+    Local l0
+      Reduce aggregates=[max((#^0 * #0))]
+        Get materialize.public.t
+
+EOF
+
+
+# A case where we cannot pull the let statement up through teh join,
+# because the local l0 is correlated against the lhs of the enclosing join
+query T multiline
+EXPLAIN RAW PLAN AS TEXT FOR
+SELECT
+  *
+FROM
+  (
+    SELECT * FROM t
+  ) as r1
+  CROSS JOIN LATERAL (
+    WITH r4 as (
+      SELECT MAX(r1.a * t.a) AS m FROM t
+    )
+    SELECT *
+    FROM
+      r4
+      CROSS JOIN LATERAL (
+        WITH r2 as (
+          SELECT MAX(r1.a * t.a) AS m FROM t
+        )
+        SELECT * FROM r2 WHERE r1.a = r4.m AND r2.m > 5
+      ) as r3
+    WHERE a != r1.a
+  ) as r5;
+----
+InnerJoin true
+  Get materialize.public.t
+  Let
+    Filter (#^0 != #^0)
+      InnerJoin true
+        Get l0
+        Let
+          Filter ((#^^0 = #^0) && (#0 > 5))
+            Get l1
+          Local l1
+            Reduce aggregates=[max((#^^0 * #0))]
+              Get materialize.public.t
```

Comment:
I know you are planning on opening a follow-up PR for `Let` output, but my two cents are that this should either be 
```
Filter ((#^^0 = #^0) && (#0 > 5))
  Reduce aggregates=[max((#^^0 * #0))]
     Get materialize.public.t
```
or 
```
InnerJoin
...
Locals
  l1
 ...
  l0
  ...
```

Maybe the number of lines involved in the local should determine whether or not they should inlined.

Reply:
I'm happy to inline a let with a single use as a transform that happens in the preparation stage.

I can certainly do the second suggestion right now.